File: contest-management-client-server-cs/networking/dto/CompetitionCountDTO.cs
#nullable disable
using System;
using model;

namespace networking.dto
{
    [Serializable]
    public class CompetitionCountDTO
    {
        public long CompetitionId { get; set; }

        public string CompetitionType { get; set; }

        public string AgeCategory { get; set; }

        public int Count { get; set; }

        public static explicit operator (Competition, int)(CompetitionCountDTO ccDto)
        {
            return
            (
                new Competition
                {
                    Id = ccDto.CompetitionId,
                    CompetitionType = ccDto.CompetitionType,
                    AgeCategory = ccDto.AgeCategory
                }, 
                ccDto.Count
            );
        }

        public static explicit operator CompetitionCountDTO((Competition, int) cc)
        {
            var (comp, count) = cc;
            return new CompetitionCountDTO
            {
                CompetitionId = comp.Id,
                CompetitionType = comp.CompetitionType,
                AgeCategory = comp.AgeCategory,
                Count = count
            };
        }
    }
}

File: contest-management-client-server-java/ContestMgmtNetworking/src/main/java/contestmgmt/networking/dto/CompetitionCountDTO.java
package contestmgmt.networking.dto;

import java.io.Serializable;

public class CompetitionCountDTO implements Serializable {
    private Long id;

    private String competitionType;

    private String ageCategory;

    private int count;

    public CompetitionCountDTO(Long id, String competitionType, String ageCategory, int count) {
        this.id = id;
        this.competitionType = competitionType;
        this.ageCategory = ageCategory;
        this.count = count;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getCompetitionType() {
        return competitionType;
    }

    public void setCompetitionType(String competitionType) {
        this.competitionType = competitionType;
    }

    public String getAgeCategory() {
        return ageCategory;
    }

    public void setAgeCategory(String ageCategory) {
        this.ageCategory = ageCategory;
    }

    public int getCount() {
        return count;
    }

    public void setCount(int count) {
        this.count = count;
    }

    @Override
    public String toString() {
        return "CompetitionDTO { id: %s, competitionType: %s, ageCategory: %s, count: %d }".formatted(id.toString(), competitionType, ageCategory, count);
    }
}
